text/0000-user-ids.md

```diff
@@ -0,0 +1,42 @@
+# Secure runtime environments
+
+## Summary
+
+This RFC recommends that Paketo Buildpacks should in general strive to avoid making run-time modifications to the Buildpacks layers or the application directory by default. Instead, they should consider the output image to be read-only by default and explicitly set the parts that need to be writeable during runtime.
+
+
+## Motivation
+
+As of Buildpacks [RFC 0085](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md) the Cloud Native Buildpacks project recommends that the run-time and build-time user ids be different. The motivation behind this RFC was to improve the security of output images and prevent arbitrary runtime modifications to built artifacts.
+
+The run-time and build-time user can potentially share the same group id if they wish to make modifications to the output image during run-time, by changing the permissions of the files they wish to modify during run-time to be group writable during the build process. Ideally most buildpacks should use the temporary directories or other known scratch directories for ephemeral runtime files.
+
+## Detailed Explanation
+
+This RFC recommends that we identify buildpacks that making runtime modifications to the application directory or buildpack layers and modify them to instead use a temporary directory - or have them make these files group writable instead. All the current Paketo stacks' build-time and run-time users belong to a unique group so changing the file permissions to be group writable should have no additional security implications for users using the Paketo stacks.
```

Comment:
For ephemeral config files and the like I think we should explicitly prefer to write to a temporary directory so that the app container can run with a read only root filesystem. If we must write to the root filesystem we should (in addition to setting the group level permissions) ensure that, whenever possible, the location is configurable with a run-time environment variable so that there is a workaround.